### PR TITLE
#165401886 Reported Article

### PR DESCRIPTION
**Pull request by sojida**
#### Description
Currently, when an article is reported, admin and reviewers cannot review the reported article. This PR implements the feature whereby an admin and a reviewer can review the reported article.

#### Type of change

- [x] New Feature(Create routes to review a reported article)

#### How Has This Been Tested?

- [x] End-to-end testing

#### Checklist:
- [x] Reviewer can review an article
- [x] Admin can change status on an article

#### PT-ID

[#165401886](https://www.pivotaltracker.com/story/show/165401886)

#### Screenshots
Screenshot 2019-04-16 at 10 45 21 PM
Screenshot 2019-04-18 at 1 53 06 PM
Screenshot 2019-04-18 at 2 36 34 PM

### CODE REVIEW

File: server/middlewares/article.middlewares.js [hunk 1]
```diff
@@ -97,11 +97,99 @@ const validateHighlight = (req, res, next) => {
...
+      errors: {
+        body: ['id not valid'],
+      },
+    });
+  }
+
+  if (!req.user.userObj.isAdmin) {
+    return res.status(401).json({
+      errors: {
+        body: ['User is not an admin'],
```
Comment (rotimi-babalola) [inline server/middlewares/article.middlewares.js:159]: This should return a 403 error code @sojida

Comment (rotimi-babalola) [reply server/middlewares/article.middlewares.js:159]: Although it depends on the action - if an authenticated user is trying to do something that only admins should do then it should definitely be a 403

File: server/middlewares/article.middlewares.js [hunk 2]
```diff
@@ -97,11 +97,99 @@ const validateHighlight = (req, res, next) => {
...
+  }
+
+  const status = adminStatus.toLowerCase();
+  if (!(status === 'accepted' || status === 'rejected')) {
+    return res.status(400).json({
+      errors: {
+        body: ['Status can only be accepted or rejected'],
+      },
+    });
+  }
```
Comment (rotimi-babalola) [inline server/middlewares/article.middlewares.js:174]: I saw a similar check like this inside the `article.controller` lines 309-311 any reason we are doing this twice? @sojida

File: server/middlewares/article.middlewares.js (hunk 1 above)
Comment (sojida) [reply server/middlewares/article.middlewares.js:159]: It is definitely a 403. I'll update this ASAP.

File: server/middlewares/article.middlewares.js (hunk 2 above)
Comment (sojida) [reply server/middlewares/article.middlewares.js:174]: Yes. In the middleware, we are checking the `req.body.status` to accept only `accepted` and `rejected`. In the controller, after we have found the reported article from the DB, we are checking if the reported article has already been given a status of `accepted` or `rejected`. We want the admin to make the action once.

File: server/controllers/article.controllers.js [hunk 3]
```diff
@@ -255,13 +255,101 @@ const createHighlight = async (req, res) => {
...
+  return res.status(200).json({
+    updatedReview,
+  });
+};
+
+const articleStatus = async (req, res) => {
+  try {
+    const { articleId } = req.params;
+    const reviewedArticle = await ReportedArticle.findOne({
+      where: { reported_article_id: articleId },
```
Comment (rotimi-babalola) [inline server/controllers/article.controllers.js]: Add a where statement to check for the article id and also where the status is `reviewed`

File: server/controllers/article.controllers.js [hunk 4]
```diff
@@ -295,6 +314,126 @@ const getUserArticles = async (req, res) => {
...
+    );
+
+    const updatedReview = await reviewedArticle.update({
+      status,
+      admin_comment: adminComment,
+    });
+    const messageBody = `<p>your article <b>${
+      reviewedArticle.article.title
+    }</b> has been reviewed and the status is <b>${status}</b></p>
+    <p>Please not that you have to unpublish this article within 2day</p>`;
```
Comment (rotimi-babalola) [inline server/controllers/article.controllers.js]: ```... within 2 days``` @sojida

File: server/controllers/article.controllers.js [hunk 5]
```diff
@@ -295,6 +314,126 @@ const getUserArticles = async (req, res) => {
...
+    );
+
+    const updatedReview = await reviewedArticle.update({
+      status,
+      admin_comment: adminComment,
+    });
+    const messageBody = `<p>your article <b>${
+      reviewedArticle.article.title
+    }</b> has been reviewed and the status is <b>${status}</b></p>
+    <p>Please not that you have to unpublish this article within 2days</p>`;
```
Comment (rotimi-babalola) [inline server/controllers/article.controllers.js]: Add a space between `2` and `days` @sojida

File: server/controllers/article.controllers.js [hunk 6]
```diff
@@ -160,37 +162,54 @@ const deleteArticle = async (req, res) => {
...
     });
 
-    // once a user is reported, then the user should not be able to review an article
-    await User.update(
-      { is_reviewer: false, is_reported: true },
-      { where: { id: reportedUserId } }
+    const messageBody = `<p>your article <b>${
+      reportedArticle.title
+    }</b> has been reported, and it is going through a review process</p>
+    <p>We will notify you when it is reviewed. Please bear with up</p>`;
```
Comment (rotimi-babalola) [inline server/controllers/article.controllers.js]: Should `us` not `up` @sojida

File: server/controllers/article.controllers.js [hunk 7]
```diff
@@ -295,6 +314,126 @@ const getUserArticles = async (req, res) => {
...
+    );
+
+    const updatedReview = await reviewedArticle.update({
+      status,
+      admin_comment: adminComment,
+    });
+    const messageBody = `<p>your article <b>${
+      reviewedArticle.article.title
+    }</b> has been reviewed and the status is <b>${status}</b></p>
+    <p>Please not that you have to unpublish this article within 2 days</p>`;
```
Comment (rotimi-babalola) [inline server/controllers/article.controllers.js]: Should be `note` not `not` @sojida

**Pull request merged**